test/functionalTest/cases/3727_avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself/notifyOnMetadataChange_true_in_condition.test

```diff
@@ -0,0 +1,256 @@
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. POST /v2/entities, to create entity E2 with attr A1 and metadata M1
+# 02. POST /v2/subscriptions, to create subscription for E2 with notifyOnMetadataChange=true
+# 03. PATCH /v2/entities/E2/attrs, to update the value of metadata
+# 04. Dump the accumulator, see the notification when metadata changes
+# 05. PATCH /v2/entities/E2/attrs, to update the value of attribute
+# 06. Dump the accumulator, see the notification when value of attribute changes
+#
+
+
+echo "01. POST /v2/entities, to create entity E2 with attr A1 and metadata M1"
+echo "======================================================================="
+payload='{
+  "id": "E2",
+  "type": "Thing",
+  "A1": {
+    "value": 26.5,
+    "type": "Float",
+    "metadata": {
+      "M1": {
+        "value": 0.77,
+        "type": "Float"
+      }
+    }
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "02. POST /v2/subscriptions, to create subscription for E2 with notifyOnMetadataChange=true"
+echo "=========================================================================================="
+payload='{
+  "description": "A subscription to get info about E2",
+  "subject": {
+    "entities": [
+      {
+        "id": "E2",
+        "type": "Thing"
+      }
+    ],
+    "condition": {
+      "attrs": [
+        "A1"
+      ]
+    }
+  },
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [
+      "A1"
+    ]
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+subId=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+echo
+echo
+
+
+echo "03. PATCH /v2/entities/E2/attrs, to update the value of metadata"
+echo "================================================================"
+payload='{
+  "A1": {
+    "metadata": {
+      "M1": {
+        "value": 1.55,
+        "type": "Float"
+      }
+    }
+  }
+}'
```

Comment:
I'd sugges to use in step 03 the same used in step 01 for the attribute value and type:
I'd suggest to use in step 03 the same used in step 01 for the attribute value and type (see my previous comment)

```
    "value": 26.5,
    "type": "Float",
```